tests/testsuite/features.rs

```diff
@@ -1730,3 +1730,35 @@ fn feature_off_dylib() {
     // Check that building without `f1` uses a dylib without `f1`.
     p.cargo("run -p bar").run();
 }
+
+#[test]
+fn warn_if_default_features() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            [project]
+            name = "foo"
+            version = "0.0.1"
+            authors = []
+
+            [dependencies.bar]
+            path = "bar"
+            optional = true
+
+            [features]
+            default-features = ["bar"]
+         "#
+        ).file("src/main.rs", "fn main() {}")
+        .file("bar/Cargo.toml",&basic_manifest("bar", "0.0.1"))
+        .file("bar/src/lib.rs", "pub fn bar() {}")
+        .build();
+
+    p.cargo("build")
+        .with_stderr("\
+[WARNING] `default-features = [\"..\"]` was found in [features]. Did you mean to use `default = [\"..\"]`?
```

Comment:
You can use the `r#"...."#` syntax (like for Cargo.toml above) to avoid having to escape the quotes.

You could in the warn emitting code too, but then you wouldn't be able to wrap the line like you are.

Just an FYI, not a problem per se :)

Reply:
Thanks! I had decided to use the regular string literal so that I could put the [WARNING] bit on a new line and keep it aligned with the rest. Otherwise it would look like r#"[WARNING]... Is there a way to do line continuation in raw strings?

Reply:
I might be misremembering but I think we just trim the front and back of the string so precise whitespace and newlines differences don't matter.

Reply:
It doesn't look like it. I just added a newline at the start and with_stderr tried to match the emitted warning with the newline.

Reply:
I guess I could use `trim()`.